File: benchmarks/decision-trees/DT4/DT4_ind.py
from sppl.compilers.sppl_to_python import SPPL_Compiler
from sppl.compilers.ast_to_spe import Id

compiler = SPPL_Compiler('''
# Population model
age ~= norm(loc=38.5816, scale=186.0614)
sex ~= choice({'female': 0.3307, 'male': 0.6693})
capital_gain ~= norm(loc=1077.6488, scale=54542539.1784)

condition(sex == 'female')
condition(age > 18)

# Decision model
if capital_gain >= 7073.5:
    if age < 20:
        t ~= atomic(loc=1)
    else:
        t ~= atomic(loc=0)
else:
    t ~= atomic(loc=1)
''')

n = compiler.execute_module()
t  = Id('t')
event = (t < 0.5)
print(n.model.prob(event))

File: benchmarks/decision-trees/DT4/DT4_ind.jl
using Alea, Distributions

DFiP = DistFix{30, 2}
inf = 33554432.0
inf_neg = -33554432.0

code = @alea begin
    # Population model
    age = bitblast(DFiP, Normal(38.5816, 186.0614), 256, inf_neg, inf)
    sex = flip(0.3307)
    capital_gain = bitblast(DFiP, Normal(1077.6488, 54542539.1784), 256, inf_neg, inf)

    observe(sex)
    observe(age > DFiP(18.0))

    # Decision model
    if capital_gain >= DFiP(7073.5)
        if age < DFiP(20.0)
            t = DFiP(1.0)
        else
            t = DFiP(0.0)
        end
    else
        t = DFiP(1.0)
    end

    result = t < DFiP(0.5)
    result
end

@show pr(code)